### PR TITLE
Added optional extra headers to getAccessToken

### PR DESCRIPTION
**Pull request by glena**
This PR provides an easy way to add custom headers on the token exchange flow.

### CODE REVIEW

**Comment by glena** [comment]
Hi @adoy, what do you think about this?
Is there any chance to get it merged?

File: src/OAuth2/Client.php [hunk 1]
```diff
@@ -205,9 +205,10 @@ public function getAuthenticationUrl($auth_endpoint, $redirect_uri, array $extra
      * @param string $token_endpoint    Url of the token endpoint
      * @param int    $grant_type        Grant Type ('authorization_code', 'password', 'client_credentials', 'refresh_token', or a custom code (@see GrantType Classes)
      * @param array  $parameters        Array sent to the server (depend on which grant type you're using)
+     * @param array  $extra_headers     Array of extra headers
      * @return array Array of parameters required by the grant_type (CF SPEC)
      */
-    public function getAccessToken($token_endpoint, $grant_type, array $parameters)
+    public function getAccessToken($token_endpoint, $grant_type, array $parameters, array $extra_headers = null)
```
Comment (adoy) [inline src/OAuth2/Client.php]: Default value could be array() so that you can assign it directly on line 227

Comment (glena) [inline src/OAuth2/Client.php]: Sure

**Comment by glena** [comment]
Cool Thanks :)

**Comment by glena** [comment]
@adoy Just one more thing, do you mind to tag the release? So I can asure people installs the right version (without using dev-master). thanks!

**Pull request merged**